server/routes/createGoal/createGoalController.ts

```diff
@@ -66,7 +66,9 @@ export default class CreateGoalController {
     }
 
     if (isEditMode(req)) {
-      // TODO - error handling for when goalIndex is not a number or goal at that index is not in the session array
+      if (!req.session.newGoals[parseInt(goalIndex, 10) - 1]) {
```

Comment:
Same as above, we could have used Error message/Error summary but theres no way the user can recover from this error themselves